woocommerce/woocommerce-gutenberg-products-block
# Store API: Introduce API Versioning System

This is a large part of #5707 which introduces versioning to the Store API, meaning endpoints can now be reached via `wc/store/v1/` as well as `wc/store/`. This does not break the existing structure which continues to map to `v1`.

What this PR does:

1. Moves all endpoints (routes and schema) under `V1` directories and PHP namespaces. Note, to maintain backwards compatibility I've maintained a list of previous namespaces which are autoloader by composer (see `deprecated.php`). This structure change was needed so that in the future, if a new version is introduced it can go under `V2` namespace and so on.
2. Registers all REST API routes (using `register_rest_route`) using `wc/store/v1` namespace. This code also registers the original `wc/store` namespace as an alias, to avoid breaking existing routes.
4. Updates unit tests to test against `v1` routes and to confirm all routes are indeed registered.
5. Updates documentation to include guiding principles and to include all updated endpoints. 

### Testing

#### Automated Tests

### Manual Testing

How to test the changes in this Pull Request:

1. Smoke test checkout process to highlight any problems with cart/checkout endpoints
2. Smoke test all products block to highlight any problems with product endpoints
3. Try accessing old endpoints, for example, `https://store.local/wp-json/wc/store/cart`. These should still work.
4. Try accessing new endpoints, for example, `https://store.local/wp-json/wc/store/v1/cart`

Unit tests cover schema and route registration.

### Changelog

> Store API - Introduced `wc/store/v1` namespace.
